README.rst

```diff
@@ -58,12 +58,164 @@ Python:
+
+Task in NI-DAQmx
+----------------
+A task is a collection of one or more virtual channels with timing, triggering, and other properties.
+Refer to the `NI-DAQmx Task <https://www.ni.com/docs/en-US/bundle/ni-daqmx/page/tasksnidaqmx.html>`_ for more information.
+
+Example code to create a task:
+
+.. code-block:: python
+
+  >>> import nidaqmx
+  >>> with nidaqmx.Task() as task:
+  ...     pass
+
+Virtual Channels in NI-DAQmx
+----------------------------
+Virtual channels, or sometimes referred to generically as channels, are software entities that encapsulate the physical channel
+along with other channel specific information—range, terminal configuration, and custom scaling—that formats the data.
+Refer to the `NI-DAQmx Channel <https://www.ni.com/docs/en-US/bundle/ni-daqmx/page/chans.html>`_ for more information.
+
+Example code to create a task to read data from an analog input channel:
+
+.. code-block:: python
+
+  >>> import nidaqmx
+  >>> with nidaqmx.Task() as task:
+  ...     ai_channel = task.ai_channels.add_ai_voltage_chan("Dev1/ai0")
+  ...     ai_channel.ai_min = -10.0
+  ...     ai_channel.ai_max = 10.0
+  ...     task.read()
+  ...
+  -0.14954069643238624
+
+Example code to create a task to read data from two analog input channels:
+
+.. code-block:: python
+
+  >>> import nidaqmx
+  >>> with nidaqmx.Task() as task:
+  ...     ai_channel_1 = task.ai_channels.add_ai_voltage_chan("Dev1/ai0")
+  ...     ai_channel_2 = task.ai_channels.add_ai_voltage_chan("Dev1/ai1")
+  ...     ai_channel_1.ai_min = -5.0
+  ...     ai_channel_1.ai_max = 5.0
+  ...     ai_channel_2.ai_min = -10.0
+  ...     ai_channel_2.ai_max = 10.0
+  ...     task.read()
+  ...
+  [-0.07477034821619312, 0.8642841883602405]
+
+Timing
+------
+You can use software timing or hardware timing to control when a signal is generated. With hardware timing,
+a digital signal, such as a clock on your device, controls the rate of generation. With software timing,
+the rate at which the samples are generated is determined by the software and operating system instead of
+by the measurement device. A hardware clock can run much faster than a software loop. A hardware clock is
+also more accurate than a software loop.
+Refer to the `Timing, Hardware Versus Software <https://www.ni.com/docs/en-US/bundle/ni-daqmx/page/hardwresoftwretiming.html>`_ for more information.
+
+Example code to acquire finite amount of data using hardware timing:
+
+.. code-block:: python
+
+  >>> import nidaqmx
+  >>> from nidaqmx.constants import AcquisitionType, READ_ALL_AVAILABLE
+  >>> with nidaqmx.Task() as task:
+  ...     task.ai_channels.add_ai_voltage_chan("Dev1/ai0")
+  ...     task.timing.cfg_samp_clk_timing(1000.0, sample_mode=AcquisitionType.FINITE, samps_per_chan=10)
+  ...     data = task.read(READ_ALL_AVAILABLE)
+  ...     print("Acquired data: [" + ", ".join(f"{value:f}" for value in data) + "]")
+  ...
+  AIChannel(name=Dev1/ai0)
```

Comment:
I don't think this is printed...

```suggestion
```

Reply:
The REPL prints return values.

Reply:
It's printed as Brad explained, I will keep it.